File: Linear_search/Haskell/linear_search.hs
linearSearch :: Eq a => [a] -> a -> Maybe Int
linearSearch xs target = search xs target 0
  where
    search [] _ _ = Nothing
    search (y:ys) t index
      | y == t = Just index
      | otherwise = search ys t (index + 1)

main :: IO ()
main = do
    let arr = [1234, 11, 73, 986, 23]
        target = 73
        res = linearSearch arr target
    case res of
        Just pos -> putStrLn $ "The element " ++ show target ++ " found at position " ++ show pos
        Nothing -> putStrLn $ "The element " ++ show target ++ " was not found"

File: Linear_search/Visual Basic Mono/linear_search.vb
Imports SystemError

Module Module1

  Sub Main()
    Dim arr() As Integer = {124, 54, 12, 872, 222}
    Dim target As Integer = 222

    Dim index As Integer = LinearSearch(arr, target)

    If index <> -1 Then
      Console.WriteLine("The element " & target & " found at position " & index)
    Else
      Console.WriteLine("The element " & target & " not found")
    End If

    Console.ReadLine()
End Sub

Function LinearSearch(arr() As Integer, target As Integer) As Integer
    Dim i As Integer
    For i = 0 To arr.Length -1
      If arr(i) = target Then
        Return i
      End If
    Next
    Return -1
End Function

End Module